### ISSUE TITLE
Import plugin for custom resource fails to re-import

### ISSUE THREAD

**Issue by securas** [godotengine/godot#46288]
May be related to #45802

**Godot version:**
3.2.3 stable
Same behavior on 3.2.4 RC2

**OS/device including version:**
Windows 10

**Issue description:**
Created an import plugin to load data from a json-like file (LDtk) and store it in a custom resource.
The resource plugin manages to import without problems.
When changing the source data on a third party editor and jumping back into godot, the reimport fails with the following report:

![](https://cdn.discordapp.com/attachments/245092069415190528/812727561536143411/unknown.png)

The data is still imported but anther behavior is observed. At least one texture is no longer re-imported.
Reimporting the texture requires restarting godot.

**Steps to reproduce:**
open example project,
make any change on the file first.ldtk on an external editor. Its a text file
jump back to godot

**Minimal reproduction project:**
[LDtk2Godot-main.zip](https://github.com/godotengine/godot/files/6017823/LDtk2Godot-main.zip)

**Comment by Picalines** [comment]
Any updates on this? I have exactly the same problem with C# in 3.3 stable release (also LDtk importer)

I wrote 2 plugins. The first one creates a TileSet resources based on the LDtk project file. The second one turns the level files into scenes.

At some stage of the import, the second plugin calls `GD.Load` to get the TileSet previously created by the first plugin. This call causes an error `BUG: File queued for import, but can't be imported`.

The main oddity is that this error only appears when you save the LDtk project and the engine reimports it. When you click on the "Reimport" button everything works stably without errors.

**Comment by securas** [comment]
AFAIK, there's no progress.

**Comment by cgbeutler** [comment]
Still happening in 3.4.4
Here was the minimal reproduction project I had created, just in case: [minimal reproduction project](https://github.com/godotengine/godot/files/8329919/bug_ImporterLoadingScripts.zip)

Anyone figure out a workaround?

**Comment by cgbeutler** [comment]
Also, the `Another resource is loaded from path '*.gd' (possible cyclic resource inclusion)` error seems to be it's own spiritic thing.
Sometimes I get it, sometimes I don't. A fun little heisenbug.
It's as if the script is loading and script cache are getting out-of-sync when used from an `EditorImportPlugin`. Maybe because the scripts are being reloaded around the same time the reimports are happening? I've been trying to make a minimal reproduction project for that error printout, but can't get it to happen reliably in a minimal reproduction.

**Comment by cgbeutler** [comment]
Seems like the `Another resource...` error to happens more consistently using a csharp-based custom resource. When this gets fixed, maybe double check with a csharp reference. Here's a minimal reproduction to help:
[bug_ImporterLoadingScripts.zip](https://github.com/godotengine/godot/files/8336853/bug_ImporterLoadingScripts.zip)
To use this minimal reproduction:
1. Open the project in the Godot Editor
2. Navigate to res://addons/custom_importer/test.custjson and double click it. In the inspector you will see that it imported the string just fine.
3. Open test.custjson in a text editor (while Godot is still running) and change the contents to something else.
4. Return to Godot and note the errors in the console. The text of the custom resource seems to have been updated fine, so the errors are maybe a red herring.

Also, I think I missed googling this issue initially because the errors were screenshot instead of pasted. To help others find this issue, here's the plaintext errors:
```
editor/editor_file_system.cpp:1724 - BUG: File queued for import, but can't be imported!
editor/editor_file_system.cpp:1725 - Method failed.
modules/gdscript/gdscript.cpp:552 - Condition "!p_keep_state && has_instances" is true. Returned: ERR_ALREADY_IN_USE
```

**Comment by Razoric480** [comment]
> Also, the `Another resource is loaded from path '*.gd' (possible cyclic resource inclusion)` error seems to be it's own spiritic thing. Sometimes I get it, sometimes I don't. A fun little heisenbug. It's as if the script is loading and script cache are getting out-of-sync when used from an `EditorImportPlugin`. Maybe because the scripts are being reloaded around the same time the reimports are happening? I've been trying to make a minimal reproduction project for that error printout, but can't get it to happen reliably in a minimal reproduction.

1. Godot scans and finds that the file needs to be updated
2. It adds it to the list and finds a valid importer
3. It then searches for dependencies.
4. For a custom resource that uses a class, it inevitably finds a GDScript file and adds that to the list.
5. It does the reimport for the original file. So far so good.
6. It tries doing the reimport for the GDScript file, but GDScript files don't have importers, so it fails.
7. Because it fails, the importer is null.
8. Because the importer is null, it triggers the ERR_FAIL macro with "BUG: File queued for import, but can't be imported!"

**Comment by AnidemDex** [comment]
Still happens for 3.4.4. I'll try to implement Razoric workaround in https://github.com/AnidemDex/Godot-CustomResource

**Comment by akien-mga** [comment]
I confirmed the fix from #67664. Will be cherry-picked for 3.6 and 3.5.2.
